### DISCUSSION TITLE
Possible to get running on Windows (no WSL)?

### DISCUSSION THREAD

**Discussion by ghost** [3rd/image.nvim#93] [Q&A] [answered]
I was hoping to get this working on Windows with Wezterm, which supports Kitty protocol, but I can't get image.nvim to load.

```
E5108: Error executing lua ...Local/nvim-data/lazy/image.nvim/lua/image/utils/term.lua:34: cannot resolve symbol 'ioctl':
 The specified procedure could not be found.

stack traceback:
        [C]: in function '__index'
        ...Local/nvim-data/lazy/image.nvim/lua/image/utils/term.lua:34: in function 'update_size'
        ...Local/nvim-data/lazy/image.nvim/lua/image/utils/term.lua:46: in main chunk
        [C]: in function 'require'
        ...Local/nvim-data/lazy/image.nvim/lua/image/utils/init.lua:7: in main chunk
        [C]: in function 'require'
        ...a/Local/nvim-data/lazy/image.nvim/lua/image/renderer.lua:2: in main chunk
        [C]: in function 'require'
        ...Data/Local/nvim-data/lazy/image.nvim/lua/image/image.lua:2: in main chunk
        [C]: in function 'require'
        ...pData/Local/nvim-data/lazy/image.nvim/lua/image/init.lua:1: in main chunk
        [C]: in function 'require'
        [string ":lua"]:1: in main chunk
Press ENTER or type command to continue
```

I have lua-for-windows, luarocks (with magick rock), and imagemagick installed under scoop.

My lazy spec for image is 

```lua
{
      '3rd/image.nvim',
      config = function ()
        package.path = package.path .. ";" .. vim.fn.expand("$HOME") .. "/scoop/apps/lua-for-windows/rocks/?/init.lua;"
        package.path = package.path .. ";" .. vim.fn.expand("$HOME") .. "/scoop/apps/lua-for-windows/rocks/?.lua;"
        require('image').setup()
      end
}
```

Does anyone know if I need to do something else, or is the plugin just not going to work on Windows?

**Comment by 3rd** [comment, marked as answer]
Hello, it won't work on Windows at all, there's no existing/planned support.
WezTerm's implementation of the graphics protocol is also not great and it's not a good mix with this plugin, the performance is horrible.
Everything is great on Linux with Kitty though, come join the dark side!

**Reply by ghost** [reply]
Oh well. I knew it was a longshot 😢. This was for my work computer; I just got image.nvim working with quarto and molten-nvim on my personal computer, and it's great!

**Reply by bew** [reply]
🤔 What's wrong with Wezterm's implementation of kitty's graphics? (or iterm2's graphics)

I'm also on Windows for work (unfortunately..), and wanted to try image.nvim out when my config's ready for that

**Reply by jgregoire** [reply]
Dang. Obligate Windows user here. Same error when running neovim natively in WezTerm.

If I try it in WSL, image.nvim reserves the vertical space for the image but it's not rendered. So close...

I guess I'm finally going to try this, so that I can use Kitty: https://learn.microsoft.com/en-us/windows/wsl/tutorials/gui-apps

**Reply by jgregoire** [reply]
Holy heck it works.

![Capture](https://github.com/3rd/image.nvim/assets/915427/0e01cf47-f2ac-4020-bff9-eeb6e413f7f7)

**Reply by 3rd** [reply]
Oh wow, that's  amazing news! How's the scroll performance?

**Reply by jgregoire** [reply]
Seems fine!

**Reply by jgregoire** [reply]
Notes for anybody who finds this and tries it:

WSL2, Windows 10 build 19045, Ubuntu 22.04. I had to install mesa in WSL before it worked.

I followed these guides:
* https://learn.microsoft.com/en-us/windows/wsl/tutorials/gui-apps
* https://github.com/danielbisar/settings/blob/main/guides/kitty-on-windows-with-wsl2.md

The kitty window is not decorated and does not tile correctly. Other programs (I tested gnome-text-editor) do not have this issue. If I find a solution, I will share it.

Edit: Fixed windowing behavior with `$ kitty -o linux_display_server=x11` It was defaulting to Wayland.

![kitty_with_images](https://github.com/3rd/image.nvim/assets/915427/9ea9f7db-2b8d-426f-87fd-4c353698d611)
